### PR TITLE
Issue #6619: Update cached Google Style Guide

### LINKED ISSUE #6619: Cached Google Style is slightly outdated and not like original
If you compare http://google.github.io/styleguide/javaguide.html and https://checkstyle.org/styleguides/google-java-style-20170228.html, you that it is displayed differently, for example original
grafik
vs. cached
grafik

Also the cached version got some minor changes that as far as I have seen do not change meaning, just add some clarification, as can be seen here: https://​github.com/google/styleguide/compare/594d91bfbb8dfcbcac0f148af8175f2b7ec9857b...91d6e367e384b0d8aaaf7ce95029514fcdf38651#diff-b6c9191ee5ccb7a5ef170e98e51d94d8

I'd like to suggest to update the cached version with a recent one that contains latest changes and also displays correctly.

### PR DESCRIPTION
**Pull request by Vampire**
Fixes #6619

### CODE REVIEW

**Comment by romani** [comment]
lets finish discussion in issue first.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  * Comments are indented at the same level as the surrounding code.
  * Detailed info about such convention can be found
  * <a href=
- * "https://checkstyle.org/styleguides/google-java-style-20170228.html#s4.8.6.1-block-comment-style">
+ * "https://checkstyle.org/styleguides/google-java-style-20180523/google.github.io/styleguide/javaguide.html#s4.8.6.1-block-comment-style">
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java]: I am not seeing `google-java-style-20180523.html`.
I assume `javaguide` is the `google-java-style-20180523`? What is the purpose of putting in such a weird nested folder instead of where it was before?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java]: @Vampire , please rebase on latest and explain extra folders.
I think you simply copied them from github repository (previously I did "page save" in browser, as it was simpler at that time, I do not remember why), so it is ok, we just need to know.

**Comment by romani** [comment]
@Vampire , your update is awesome step forward, please find time to finish it.

**Comment by Vampire** [comment]
> Please rebase on the latest master

Done

> and add new label to google style guide (similar to sun) on items discussed in main issue that need to still be updated.

Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java (hunk 1 above)
Comment (Vampire) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java]: > I am not seeing `google-java-style-20180523.html`.
> I assume `javaguide` is the `google-java-style-20180523`?

Exactly

> What is the purpose of putting in such a weird nested folder instead of where it was before?

Before @romani just used the browsers "Save page as" functionality, which most often does not work as intended and for example added these ugly effects I highlighted in the issue, as the page-save saves stuff that was added dynamically via JavaScript. But the JavaScript is also still there, so things are doubled. And the dynamically added images are not where expected, so you get those missing images displayed.

I used the software "WinHTTrack Website Copier" to do a proper site download that displays correctly, including all resources where they are expected.

**Comment by Vampire** [comment]
The failing AppVeyor build has nothing to do with my changes.
It seems the build cannot download the referenced JGit version from Maven Central.

**Comment by Vampire** [comment]
Well, my intention was to keep it like the website copier left it for easy updating and diffing in the future.
If you guys really have a problem with the paths, we can change that of course.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java]: > I used the software "WinHTTrack Website Copier"

After some time of thinking on this ...
lets not depend on some tool here, I do not know such tool, and I will not be able to use it as it is  for Windows.
Please update please folders to be at `google-java-style-20180523` I hope style guide html style will not be changed in recent few years. So all we will need to update periodically is html file. So some any tool, and even "Save as" will work.

File: src/xdocs/google_style.xml [hunk 2]
```diff
@@ -21,10 +21,10 @@
                 <p>
                     This coverage report was created for
                     <a href="http://google.github.io/styleguide/javaguide.html">
-                        Google Java Style</a>(
-                    <a href="styleguides/google-java-style-20170228.html">
-                          cached page</a>)
-                    , version of 28 February 2017
+                        Google Java Style</a>(<a
```
Comment (romani) [inline src/xdocs/google_style.xml]: please restore formatting here, I like `a` to start a line.

File: src/xdocs/google_style.xml [hunk 3]
```diff
@@ -21,10 +21,10 @@
...
+                        href="styleguides/google-java-style-20180523/google.github.io/styleguide/javaguide.html"
+                    >cached page</a>),
+                    version of 23 May 2018, current as of 30 March 2019
```
Comment (romani) [inline src/xdocs/google_style.xml]: I am not sure why we need "30 March", what is reason to have it ?
Not sure why user need to know date of export. We use date as some sort of anchor to commit in googe style repo.
it is probably better to place  SHA1 of commit/link-to-commit to be even more exact.

File: src/xdocs/google_style.xml [hunk 4]
```diff
@@ -909,36 +914,43 @@
...
                                 <br />
                                 <br />
-                                <img src="images/ok_green.png" alt="" />
+                                <img src="images/ok_blue.png" alt="" />
                                 <a
                                     href="config_whitespace.html#MethodParamPad">MethodParamPad</a>
                                 <br />
                                 <br />
-                                <img src="images/ok_green.png" alt="" />
+                                <img src="images/ok_blue.png" alt="" />
```
Comment (romani) [inline src/xdocs/google_style.xml]: lets keep all non-NoWhitespaceBefore as green because "Existing Check covers all requirements from Google.".

File: src/xdocs/style_configs.xml [hunk 5]
```diff
@@ -21,8 +21,8 @@
       </p>
 
       <ul>
-        <li><a href="google_style.html">Google's style</a>;</li>
-        <li><a href="sun_style.html">Sun's style</a>.</li>
+        <li><a href="google_style.html">Google's style</a></li>
+        <li><a href="sun_style.html">Sun's style</a></li>
```
Comment (romani) [inline src/xdocs/style_configs.xml]: it would be better to move this change to separate commit on this PR as fix is not related to issue.
you can use commit message prefix "minor: ......"
We usually use old commits as reference to amount of updates required for some periodical updates. So better to split unrelated changes in different commits.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java (hunk 1 above)
Comment (Vampire) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java]: > lets not depend on some tool here, I do not know such tool, and I will not be able to use it as it is for Windows.

It is not.
Don't get trapped by the "Win" in the name, it is just the Windows build of HTTrack which is available for Windows, Linux, OSX and even Android: http://www.httrack.com/page/2/

> Please update please folders to be at `google-java-style-20180523` I hope style guide html style will not be changed in recent few years. So all we will need to update periodically is html file. So some any tool, and even "Save as" will work.

The nice thing with HTTrack is, that it properly downloads websites and all related things that are necessary for offline or cached version.
Actually in this case it is not necessary yes, because it the actual HTML that is stored in the Google repository, so we can just copy in the files.
But keep it in mind for probably future supported style guides where the source is not the actual HTML, there HTTrack or any other proper website copier is much better and more reliable than using "Save as" in the browser which will usually save JS-post-processed versions of the pages which end up unnice like it was with the Google Styleguide here too.

File: src/xdocs/google_style.xml (hunk 2 above)
Comment (Vampire) [reply src/xdocs/google_style.xml]: In source view I'd agree.
But here I cared more about the result.
Your version:
grafik
My version:
grafik
Like it was before, there are spurious spaces that imho shouldn't be there, which is why I moved the beginning of the `<a>` and the `comma` to the end of the previous line.
Don't you think the result is more important to be nice here?

File: src/xdocs/google_style.xml (hunk 3 above)
Comment (Vampire) [reply src/xdocs/google_style.xml]: Well, I as user would be interested in this information tbh.
It is more visible with the sun style where we have "version of 20 April 1999, current as of 30 March 2019".
It shows the user when was last checked that the cached and reported-on version is current.
If I only read "version of 20 April 1999" or "version of 23 May 2018", I ask myself "is this also compatible with the latest version, were there any changes?".
If I see that it was last checked on "30 March 2019", I'm pretty confident the report is against a pretty recent version of the styleguide.
(Updated to 07 May 2019 now, as I just rechecked it is still recent)

File: src/xdocs/google_style.xml (hunk 4 above)
Comment (Vampire) [reply src/xdocs/google_style.xml]: Well, I interpreted it as "the combination of these checks covers all requirements.
Because actually none of the single checks "covers all requirements".
That's why I switched them all to blue and maybe there should actually be only one icon per styleguide rule, not one icon per checkstyle rule.
For now I switched the others back to green as requested.

File: src/xdocs/style_configs.xml (hunk 5 above)
Comment (Vampire) [reply src/xdocs/style_configs.xml]: fixed, all updates pushed

File: src/xdocs/google_style.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/google_style.xml]: Thanks, for explanation, let's keep your update
